scala-package/mxnet-demo/scala-demo/Makefile

```diff
@@ -44,11 +43,14 @@ endif
 scalademo:
 	(mvn package -Dmxnet.profile=$(SCALA_PKG_PROFILE) \
 		-Dmxnet.scalaprofile=$(SCALA_VERSION_PROFILE) \
-		-Dmxnet.version=$(MXNET_VERSION) \
+		-Dscala.version=$(SCALA_VERSION))
+
+scalacidemo:
```

Comment:
Two comments:
1.) (Nitpick) I'm not crazy about this name, mainly because scalac is the compiler and this is easily read as "scalac i demo".
2.) This is being done a bit differently than how the java-demo one is done. First, the Java one only has javademo and javaclean. The nightly uses make javademo instead of introducing a javacidemo. Also, the javademo makefile uses mvn install instead of mvn package. Are you aware of anything that keeps the two from being consistent with each other?

Reply:
I think the differences in java-demo and scala-demo projects are only because they were developed independent of each other and over two different periods of time. 

I'll take your suggestion here and make both of them consistent because essentially they are just showing how to setup a sample project with jars pulled in from Maven. 

Will standardize java-demo, scala-demo in this PR itself.